### PR TITLE
Always suppress incoming offers when task queue is empty

### PR DESCRIPTION
**Pull request by huadongliu**
This is to deal with leader re-election. It does not hurt on normal conditions.

### CODE REVIEW

File: task_processing/plugins/mesos/execution_framework.py [hunk 1]
```diff
@@ -423,16 +423,17 @@ def resourceOffers(self, driver, offers) -> None:
 
         with self._lock:
             if self.task_queue.empty():
-                if not self.are_offers_suppressed:
```
Comment (sagar8192) [inline task_processing/plugins/mesos/execution_framework.py]: You can get rid of this variable now, right?

Comment (huadongliu) [reply task_processing/plugins/mesos/execution_framework.py]: are_offers_suppressed is still used in enqueue_task.

Comment (sagar8192) [reply task_processing/plugins/mesos/execution_framework.py]: Ah! ship-it

**Comment by keymone** [comment]
do we not get any notification from mesos that leader election happened?

**Pull request merged**